### PR TITLE
Added rubocop, fixed some offenses and added rubocop_todo

### PR DESCRIPTION
**Pull request by frankyston**
Rubocop has been added to improve the code quality of the project.

There is a file called `rubocop_todo` to fix the offenses gradually.

It was also added to Circle CI

### CODE REVIEW

**Comment by frankyston** [comment]
Issue: https://github.com/call4paperz/call4paperz/issues/346

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -8,9 +8,7 @@ class ApplicationController < ActionController::Base
 
 protected
   def store_location
-    if (request.get? && request.format.html? && !request.xhr? && !devise_controller?)
-      store_location_for(:user, request.url)
-    end
+    store_location_for(:user, request.url) if (request.get? && request.format.html? && !request.xhr? && !devise_controller?)
```
Comment (diegopolido) [inline app/controllers/application_controller.rb]: I would used guard clause instead

Comment (diegopolido) [inline app/controllers/application_controller.rb]:
```suggestion
    return unless (request.get? && request.format.html? && !request.xhr? && !devise_controller?)
    
    store_location_for(:user, request.url)
```

Comment (diegopolido) [reply app/controllers/application_controller.rb]: or even many guard clauses

File: app/controllers/proposals_controller.rb [hunk 2]
```diff
@@ -96,9 +96,7 @@ def verify_event_closed
   end
 
   def verify_grace_period
-    unless proposal.has_grace_period_left?
-      redirect_to [event, proposal], :notice => "You cannot edit a proposal after 30 minutes of creation."
-    end
+    redirect_to [event, proposal], :notice => "You cannot edit a proposal after 30 minutes of creation." unless proposal.has_grace_period_left?
```
Comment (diegopolido) [inline app/controllers/proposals_controller.rb]:
```suggestion
    return if proposal.has_grace_period_left?
    
    redirect_to [event, proposal], :notice => "You cannot edit a proposal after 30 minutes of creation."
```

**Comment by lucasrenan** [comment]
@frankyston we can also add it on code climate https://docs.codeclimate.com/docs/rubocop

**Comment by frankyston** [comment]
> @frankyston we can also add it on code climate https://docs.codeclimate.com/docs/rubocop

Cool, I didn't know you could put it in, I'll remove from circle ci and add in code climate

**Comment by lucasrenan** [comment]
@frankyston I added for you here: https://github.com/call4paperz/call4paperz/pull/354

**Comment by frankyston** [comment]
@lucasrenan removed rubocop

File: Gemfile [hunk 3]
```diff
@@ -47,6 +47,7 @@ group :development, :test do
   gem 'rspec-rails'
   gem 'dotenv-rails'
   gem 'pry-byebug'
+  gem 'rubocop', require: false
```
Comment (lucasrenan) [inline Gemfile]: we should use `0.74`, as it's the most recent version supported by code climate.

**Comment by lucianosousa** [comment]
@frankyston can you please merge master here first?
Build is failing and I guess that's the reason

**Comment by frankyston** [comment]
> @frankyston can you please merge master here first?
> Build is failing and I guess that's the reason

I performed the merge of the master but continues the error in the build, I'm investigating what is the problem.

```
Traceback (most recent call last):
	2: from /usr/local/bin/bundle:23:in `<main>'
	1: from /usr/local/lib/ruby/2.6.0/rubygems.rb:302:in `activate_bin_path'
/usr/local/lib/ruby/2.6.0/rubygems.rb:283:in `find_spec_for_exe': Could not find 'bundler' (2.0.2) required by your /home/circleci/repo/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:2.0.2`
Exited with code 1
```

**Comment by frankyston** [comment]
For some reason the builds are running on ruby 2.6.0 and it is set to run on ruby 2.6.5, I think it has something to do with it.

File: spec/controllers/twitter_controller_spec.rb [hunk 4]
```diff
@@ -15,7 +15,7 @@
   end
 
   describe "last tweets" do
-    it "should get last tweet from call4paperz timeline" do
+    xit "should get last tweet from call4paperz timeline" do
```
Comment (lucianosousa) [inline spec/controllers/twitter_controller_spec.rb]: why?

Comment (frankyston) [reply spec/controllers/twitter_controller_spec.rb]: If you have any tips, I'm accepting 😄

**Pull request closed without merging**